libs/openFrameworks/types/ofBaseTypes.h

```diff
@@ -621,218 +621,1051 @@ class ofBaseVideoPlayer: virtual public ofBaseVideo{
+	/// \brief Draw an ellipse using a 3D point, width, and height.
+	/// \param x The x coordinate of the center of the circle.
+	/// \param y The y coordinate of the center of the circle.
+	/// \param z The z coordinate of the center of the circle.
+	/// \param width The width of the circle.
+	/// \param height The height of the circle.
 	virtual void drawEllipse(float x, float y, float z, float width, float height) const=0;
+	/// \brief Draw text with this renderer using the current bitmap text mode.
+	///
+	/// When using the OF_BITMAPMODE_SIMPLE bitmap text strings are drawn with
+	/// their origin at the bottom left corner of the text. However,
+	/// setBitmapTextMode() can transform this default behavior with modes like:
+	///		OF_BITMAPMODE_SIMPLE
+	///		OF_BITMAPMODE_SCREEN
+	///		OF_BITMAPMODE_VIEWPORT
+	///		OF_BITMAPMODE_MODEL
+	///		OF_BITMAPMODE_MODEL_BILLBOARD
+	///
+	/// \param text The text to draw with the renderer.
+	/// \param x The x position for the bottom of \p text.
+	/// \param y The y position for the left alignment of \p text.
+	/// \param z The z position of the text.
 	virtual void drawString(string text, float x, float y, float z) const=0;
+	/// \brief Draw text with this renderer using an ofTrueType font.
+	/// \param font The font to use when drawing \p text.
+	/// \param text The text to draw with the renderer.
+	/// \param x The x position for the bottom of \p text.
+	/// \param y The y position for the left alignment of \p text.
 	virtual void drawString(const ofTrueTypeFont & font, string text, float x, float y) const=0;
 
 
 	// returns true an ofPath to draw with, this allows to keep
 	// immediate mode rendering working in multiwindow with multiple
 	// contexts without reimplementing the logic on every renderer
+	/// \brief Get a reference to the path used internally by this renderer.
+	/// \returns A reference to the path used internally by this renderer.
 	virtual ofPath & getPath()=0;
 
+	/// \brief Get this renderer's current style object.
+	/// \returns This renderer's current style object.
 	virtual ofStyle getStyle() const=0;
+	/// \brief Set this renderer's current style object.
+	/// \param style The style object to set this renderer to use.
 	virtual void setStyle(const ofStyle & style) = 0;
+	/// \brief Push this renderer's current style into its internal style stack.
+	///
+	/// This creates a new style object used by this renderer internally until
+	/// popStyle() is called.
+	///
 	virtual void pushStyle()=0;
+	/// \brief Pop this renderer's current style from its internal style stack.
+	///
+	/// This restores the style that was last saved with pushStyle().
+	///
 	virtual void popStyle()=0;
 
+	/// \brief Set the resolution used when drawing curves with this renderer.
+	/// \param resolution The resolution to request this renderer to use when
+	/// drawing curves.
 	virtual void setCurveResolution(int resolution)=0;
+	
+	/// \brief Set this renderer's poly winding mode.
```

Comment:
I don't understand enough about poly winding mode to give much more of a description than this, so if someone knows more please fill in :smile_cat: